proposals/numeric-intptr.md

```diff
@@ -0,0 +1,344 @@
+  ...
+  nint operator >>>(nint x, nint count);
+  nuint operator >>>(nuint x, nint count);
+  ```
+
+For the predefined operators, the number of bits to shift is computed as follows:
+\[...]
+- When the type of `x` is `nint` or `nuint`, the shift count is given by the low-order five bits of `count` on a 32 bits platform, or the lower-order six bits of `count` on a 64 bits platform.  
+The shift count is computed from `count & (sizeof(nint) * 8 - 1)`, which is `count & 0x1F` on a 32 bits platform and `count & 0x3F` on a 64 bits platform.
+
+## 11.11 Relational and type-testing operators
+
+### 11.11.2 Integer comparison operators
+
+The predefined integer comparison operators are:
+
+```csharp
+...
+bool operator ==(nint x, nint y);
+bool operator ==(nuint x, nuint y);
+
+bool operator !=(nint x, nint y);
+bool operator !=(nuint x, nuint y);
+
+bool operator <(nint x, nint y);
+bool operator <(nuint x, nuint y);
+
+bool operator >(nint x, nint y);
+bool operator >(nuint x, nuint y);
+
+bool operator <=(nint x, nint y);
+bool operator <=(nuint x, nuint y);
+
+bool operator >=(nint x, nint y);
+bool operator >=(nuint x, nuint y);
+```
+
+## 11.12 Logical operators
+
+### 11.12.2 Integer logical operators
+
+The predefined integer logical operators are:
+
+```csharp
+...
+nint operator &(nint x, nint y);
+nuint operator &(nuint x, nuint y);
+
+nint operator |(nint x, nint y);
+nuint operator |(nuint x, nuint y);
+
+nint operator ^(nint x, nint y);
+nuint operator ^(nuint x, nuint y);
+```
+
+## 11.20 Constant expressions
+
+A constant expression may be either a value type or a reference type. If a constant expression is a value type, it must be one of the following types: `sbyte`, `byte`, `short`, `ushort`, `int`, `uint`, **`nint`, `nuint`**, `long`, `ulong`, `char`, `float`, `double`, `decimal`, `bool,` or any enumeration type.
+
+\[...]
+
+An implicit constant expression conversion permits a constant expression of type `int` to be converted to `sbyte`, `byte`, `short`, `ushort`, `uint`, **`nint`, `nuint`,** or `ulong`, provided the value of the constant expression is within the range of the destination type.
+
+## 22.5 Pointer conversions
+
+### 22.5.1 General
+
+\[...]
+
+Additionally, in an unsafe context, the set of available explicit conversions is extended to include the following explicit pointer conversions:
+
+- From any *pointer_type* to any other *pointer_type*.
+- From `sbyte`, `byte`, `short`, `ushort`, `int`, `uint`, **`nint`, `nuint`,** `long`, or `ulong` to any *pointer_type*.
+- From any *pointer_type* to `sbyte`, `byte`, `short`, `ushort`, `int`, `uint`, **`nint`, `nuint`,** `long`, or `ulong`.
+
+## Various considerations
+
+### Breaking changes
+
+One of the main impacts of this design is that `System.IntPtr` and `System.UIntPtr` gain some built-in operators (conversions, unary and binary).  
+Those include `checked` operators, which means that some operators on those types will now throw when overflowing. For example:
```

Comment:
>that some operators on those types [](http://example.com/codeflow?start=47&length=34)

Perhaps "the following operators" to indicate this is the complete list of overflow changes.

Reply:
I don't think that's the complete list. How about multiplication?

Reply:
Multiplication is not available for `IntPtr` with C#10.